### PR TITLE
Recipe Remainder API

### PR DESCRIPTION
**Pull request by OroArmor**
An extensive, no missed location, API for recipe remainders.

Any crafting location has its location checked for remainders and carefully implemented. The impl also makes it easy for users to add this into their own code

### CODE REVIEW

File: library/item/item_setting/src/main/java/org/quiltmc/qsl/item/setting/api/QuiltItemSettings.java [hunk 1]
```diff
@@ -64,6 +64,55 @@ public QuiltItemSettings customDamage(CustomDamageHandler handler) {
...
+	 * @param by the amount
+	 */
+	public QuiltItemSettings damageIfUsedInCrafting(int by) {
+		if (by == 0) {
+			return this.recipeRemainder((original, recipe) -> original);
+		}
+
+		return this.recipeRemainder((original, recipe) -> {
+			if (!original.isDamageable()) {
+				return ItemStack.EMPTY;
```
Comment (Leo40Git) [inline library/item/item_setting/src/main/java/org/quiltmc/qsl/item/setting/api/QuiltItemSettings.java:100]: Should probably throw an exception or something here, just returning `ItemStack.EMPTY` will cause confusion.

File: library/item/item_setting/src/main/java/org/quiltmc/qsl/item/setting/mixin/reciperemainder/StonecutterOutputSlotMixin.java [hunk 2]
```diff
@@ -16,8 +16,8 @@
 
 package org.quiltmc.qsl.item.setting.mixin.reciperemainder;
 
-import org.quiltmc.qsl.item.setting.impl.CustomItemSettingImpl;
-import org.quiltmc.qsl.item.setting.impl.RecipeRemainderLocationHandler;
+import org.checkerframework.common.aliasing.qual.Unique;
```
Comment (Leo40Git) [inline library/item/item_setting/src/main/java/org/quiltmc/qsl/item/setting/mixin/reciperemainder/StonecutterOutputSlotMixin.java:19]: Wrong import!

**Comment by OroArmor** [comment]
I am looking for someone to take over this PR from me. I don't have time to keep working on it.

**Comment by Platymemo** [comment]
I can take over this PR

**Pull request closed without merging**